flake.nix

```diff
@@ -7,6 +7,7 @@
     darwinModules.home-manager = import ./nix-darwin;
 
     lib = {
+      hm = import ./modules/lib;
```

Comment:
Isn't it necessary to wire in the Nixpkgs `lib`? I.e., something like `hm = import ./modules/lib { lib = nixpkgs.lib; }`.

Reply:
But then you are forced to override home-manager's nixpkgs reference in your configuration and you can only set it to one value

Reply:
I'm not certain I follow. I know next to nothing about flakes but I understood that this `hm` is completely separate from the `hm` inside the `lib` passed to the HM modules? This `hm` would be used only for code outside HM modules, I imagine.

Or do entries inside the `lib` attribute set of a flake automatically extend the Nixpkgs `lib`? In that case I could imagine there being some problems with conflicting definitions when HM tries to extend `pkgs.lib`.

Reply:
A flake's definition includes `inputs` (dependencies) that are required. `nixpkgs` (and maybe others) are implicit dependencies of every flake (of course you can define your own input named `nixpkgs` to override the default behavior)
When using another flake, you can override its `inputs`.
If we pass the suggested argument to the imported function, the end user would be forced to override Home Manager's `nixpkgs` input to use the library with a specific revision of `nixpkgs`.
Another downside is that the dependency cannot be overriden more than once (I cannot use `lib.hm` with both `nixpkgs-unstable` and `nixpkgs-20.09` unless I specify Home Manager twice in my `inputs` and override each `nixpkgs`)

Reply:
> But then you are forced to override home-manager's nixpkgs reference in your configuration and you can only set it to one value

nixpkgs is already an input of home-manager. Nix will use the registry version (see `nix registry list`) if not specified when pulling `home-manager` e.g. using something like:

```nix
{
  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-20.09";
  inputs.home-manager.url =
    "github:nix-community/home-manager/release-20.09";
  inputs.home-manager.inputs.nixpkgs.follows = "nixpkgs";
  # ...
}
```

> Or do entries inside the lib attribute set of a flake automatically extend the Nixpkgs lib? In that case I could imagine there being some problems with conflicting definitions when HM tries to extend pkgs.lib.

No, AFAIK there's nothing special about the `lib` attribute.

Reply:
> Isn't it necessary to wire in the Nixpkgs lib? I.e., something like hm = import ./modules/lib { lib = nixpkgs.lib; }.

Yes it is, small derp from my side :slightly_smiling_face: 

Fixed & tested using `nix eval .#lib.hm`.